src/R2_alg.jl

```diff
@@ -69,9 +69,10 @@ function R2(
   f::F,
   ∇f!::G,
   h::H,
-  options::ROSolverOptions,
-  x0::AbstractVector,
-) where {F <: Function, G <: Function, H}
+  options::ROSolverOptions{R},
+  x0::AbstractVector{R},
+  neg_tol::R = -eps(R)^(1/3) ,
+) where {F <: Function, G <: Function, H, R <: Real}
```

Comment:
It looks good now, thank you! Now that I think about it, we should put `neg_tol` in the options. Then it would be available to the other solvers.